src/jit/gentree.h

```diff
@@ -120,6 +120,8 @@ enum genTreeKinds
     GTK_NOVALUE = 0x0400, // node does not produce a value
     GTK_NOTLIR  = 0x0800, // node is not allowed in LIR
 
+    GTK_NOCONTAIN = 0x1000, // this node is a value, but may not be contained
```

Comment:
Leaving the RA to make reg optional decisions is the right thing but in my very general hand wavy idea when the RA makes a decision to use mem rather than reg the IR would be modified for the ref position in resolution to use a MEM node for the spill.  The overarching idea is to minimize the contracts between phases and put the transformation where the decision was made, the contract then could narrowed down to a simpler, unambiguous IR (fingers crossed).  Ideally we would make Lower more responsible for shaping the IR for the target and CodeGen more responsible for encoding the bits.